hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/BlockCache.java

```diff
@@ -161,4 +164,14 @@ default Cacheable getBlock(BlockCacheKey cacheKey, boolean caching, boolean repe
   default boolean isMetaBlock(BlockType blockType) {
     return blockType != null && blockType.getCategory() != BlockType.BlockCategory.DATA;
   }
+
+  /**
+   * Clean Cache by evicting the blocks of files belonging to regions that are no longer served by
+   * the RegionServer.
+   * @param server HRegionServer
+   * @return A map of filename and number of blocks evicted.
+   */
+  default Optional<Map<String, Integer>> uncacheStaleBlocks(HRegionServer server) {
```

Comment:
I prefer here we pass an interface, for testing whether a region is available, instead of pass a HRegionServer directly.  It will be easier for testing.
And just returning a Map is enough? If we do not clean any blocks, just return an empty map?